pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -449,32 +458,32 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation, errorContext *er
 			waitUntilControlPlaneExposureDeleted,
 			destroyNginxIngressDNSRecord,
 			destroyExternalDomainDNSRecord,
+			destroyInternalDomainDNSRecord,
 			waitUntilInfrastructureDeleted,
 		)
 
-		destroyInternalDomainDNSRecord = g.Add(flow.Task{
-			Name:         "Destroying internal domain DNS record",
-			Fn:           flow.TaskFn(botanist.DestroyInternalDomainDNSRecord).DoIf(dnsEnabled),
-			Dependencies: flow.NewTaskIDs(syncPoint),
-		})
-		deleteDNSProviders = g.Add(flow.Task{
-			Name:         "Deleting DNS providers",
-			Fn:           flow.TaskFn(botanist.DeleteDNSProviders).DoIf(dnsEnabled),
-			Dependencies: flow.NewTaskIDs(destroyInternalDomainDNSRecord),
-		})
 		deleteNamespace = g.Add(flow.Task{
 			Name:         "Deleting shoot namespace in Seed",
 			Fn:           flow.TaskFn(botanist.DeleteNamespace).Retry(defaultInterval),
-			Dependencies: flow.NewTaskIDs(syncPoint, deleteDNSProviders, deleteKubeAPIServer),
+			Dependencies: flow.NewTaskIDs(syncPoint, deleteKubeAPIServer),
 		})
 		_ = g.Add(flow.Task{
 			Name:         "Waiting until shoot namespace in Seed has been deleted",
 			Fn:           botanist.WaitUntilSeedNamespaceDeleted,
 			Dependencies: flow.NewTaskIDs(deleteNamespace),
 		})
-
-		f = g.Compile()
 	)
+
+	for k, v := range botanist.Shoot.Components.DNS.Additional {
+		_ = g.Add(flow.Task{
+			Name:         fmt.Sprintf("Removing additional DNSProvider %q", k),
+			Fn:           flow.TaskFn(component.OpWaiter(v).Deploy).DoIf(wakeupRequired),
```

Comment:
Is everything as expected here? The task name is `Removing additional DNSProvider %q` but the invoked func is `Deploy`, not `Destroy`.